### PR TITLE
docs(codecs, sinks): Document changes to sink encoding

### PR DESCRIPTION
**Pull request by pablosichert**
This PR documents the changes / features that have been added with #9459.

### CODE REVIEW

File: website/cue/reference/components/sinks.cue [hunk 1]
```diff
@@ -209,9 +247,6 @@ components: sinks: [Name=string]: {
 													json: "JSON encoded event."
 												}
 											}
-											if codec == "ndjson" {
```
Comment (jszwedko) [inline website/cue/reference/components/sinks.cue:212]: Are the `if batched`s above still relevant given the introduction of the framing options?

File: website/cue/reference/components/sinks.cue [hunk 2]
```diff
@@ -221,6 +256,14 @@ components: sinks: [Name=string]: {
 							}
 						}
 						options: {
+							if codec == "avro" {
```
Comment (jszwedko) [inline website/cue/reference/components/sinks.cue:259]: I think what you want here is to check if `features.send.encoding.codec.enum` _contains_ `avro` (see cue failure).

**Review by jszwedko — COMMENTED** [review]
A couple of notes, but looking good!

File: website/content/en/highlights/2022-05-23-0-23-0-upgrade-guide.md [hunk 3]
```diff
@@ -203,10 +186,104 @@ And the `syslog` source returned:
...
+
+```
+framing.method = "newline_delimited"
+encoding.codec = "json"
+```
+
+if the sinks supports framing or as
+
+```
+encoding.codec = "json"
```
Comment (spencergilbert) [inline website/content/en/highlights/2022-05-23-0-23-0-upgrade-guide.md:286]: Which sinks "default" to _ndjson_, can we note in their docs that json == ndjson?

Comment (pablosichert) [reply website/content/en/highlights/2022-05-23-0-23-0-upgrade-guide.md:286]: Ah, that is a good point 🤔 Will figure out the exact set of sinks that are affected and either don't have a framing option or don't need to set it explicitly

Comment (pablosichert) [reply website/content/en/highlights/2022-05-23-0-23-0-upgrade-guide.md:286]: Ah, that is a good point 🤔 Will figure out the exact set of sinks that are affected and either don't have a framing option or don't need to set it explicitly

**Review by spencergilbert — APPROVED** [review]

**Pull request merged**